File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/adapter/LocationAdapter.java
package com.byagowi.persiancalendar.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.byagowi.persiancalendar.Constants;
import com.byagowi.persiancalendar.databinding.ListItemCityNameBinding;
import com.byagowi.persiancalendar.entity.CityEntity;
import com.byagowi.persiancalendar.util.Utils;
import com.byagowi.persiancalendar.view.preferences.LocationPreferenceDialog;
import com.byagowi.persiancalendar.viewmodel.LocationAdapterViewModel;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class LocationAdapter extends RecyclerView.Adapter<LocationAdapter.ViewHolder> {
    final private List<CityEntity> mCities;
    final private LocationPreferenceDialog mLocationPreferenceDialog;

    public LocationAdapter(LocationPreferenceDialog locationPreferenceDialog,
                           List<CityEntity> cities) {
        mLocationPreferenceDialog = locationPreferenceDialog;
        mCities = cities;
    }

    @NonNull
    @Override
    public LocationAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ListItemCityNameBinding binding = ListItemCityNameBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(mCities.get(position));
    }

    @Override
    public int getItemCount() {
        return mCities.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        private ListItemCityNameBinding binding;

        ViewHolder(ListItemCityNameBinding itemBinding) {
            super(itemBinding.getRoot());
            this.binding = itemBinding;
        }

        void bind(CityEntity cityEntity) {
            String city, country;
            switch (Utils.getAppLanguage()) {
                case Constants.LANG_EN_IR:
                case Constants.LANG_EN_US:
                    city = cityEntity.getEn();
                    country = cityEntity.getCountryEn();
                    break;
                case Constants.LANG_CKB:
                    city = cityEntity.getCkb();
                    country = cityEntity.getCountryCkb();
                    break;
                case Constants.LANG_AR:
                    city = cityEntity.getAr();
                    country = cityEntity.getCountryAr();
                    break;
                default:
                    city = cityEntity.getFa();
                    country = cityEntity.getCountryFa();
                    break;
            }
            LocationAdapterViewModel model = new LocationAdapterViewModel(city, country, this);
            binding.setModel(model);
            binding.executePendingBindings();
        }

        @Override
        public void onClick(View view) {
            mLocationPreferenceDialog.selectItem(mCities.get(getAdapterPosition()).getKey());
        }
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/adapter/LocationAdapter.kt
package com.byagowi.persiancalendar.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.byagowi.persiancalendar.Constants
import com.byagowi.persiancalendar.R
import com.byagowi.persiancalendar.entity.CityEntity
import com.byagowi.persiancalendar.util.Utils
import com.byagowi.persiancalendar.view.preferences.LocationPreferenceDialog
import java.util.Collections.emptyList

class LocationAdapter(private val locationPreferenceDialog: LocationPreferenceDialog) : RecyclerView.Adapter<LocationAdapter.ViewHolder>() {
  private val locale: String = Utils.appLanguage
  private val cities: List<CityEntity>
  private val layoutInflater: LayoutInflater

  init {
    val ctx = locationPreferenceDialog.context
    this.layoutInflater = LayoutInflater.from(ctx)
    this.cities = if (ctx == null) emptyList() else Utils.getAllCities(ctx, true)
  }

  inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
    val tvCountry: TextView = itemView.findViewById(R.id.tvCity)
    val tvCity: TextView = itemView.findViewById(R.id.tvCountry)

    init {
      itemView.setOnClickListener(this)
    }

    override fun onClick(view: View) =
        locationPreferenceDialog.selectItem(cities[adapterPosition].key)
  }

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocationAdapter.ViewHolder =
      ViewHolder(layoutInflater.inflate(R.layout.list_item_city_name, parent, false))

  override fun onBindViewHolder(holder: ViewHolder, position: Int) = when (locale) {
    Constants.LANG_EN -> {
      holder.tvCity.text = cities[position].en
      holder.tvCountry.text = cities[position].countryEn
    }
    Constants.LANG_CKB -> {
      holder.tvCity.text = cities[position].ckb
      holder.tvCountry.text = cities[position].countryCkb
    }
    else -> {
      holder.tvCity.text = cities[position].fa
      holder.tvCountry.text = cities[position].countryFa
    }
  }

  override fun getItemCount(): Int = cities.size
}